wagtail/contrib/redirects/views.py

```diff
@@ -127,49 +127,24 @@ def header_more_buttons(self):
         ]
 
 
-@permission_checker.require("change")
-def edit(request, redirect_id):
-    theredirect = get_object_or_404(models.Redirect, id=redirect_id)
+class Edit(generic.EditView):
+    model = models.Redirect
+    form_class = RedirectForm
+    index_url_name = "wagtailredirects:index"
+    edit_url_name = "wagtailredirects:edit"
+    delete_url_name = "wagtailredirects:delete"
+    permission_required = "change_redirect"
+    delete_item_label = _("Delete")
+    submit_button_label = _("Save")
 
-    if not permission_policy.user_has_permission_for_instance(
-        request.user, "change", theredirect
-    ):
-        raise PermissionDenied
+    def get_object(self, queryset=None):
+        redirect_id = self.kwargs.get("redirect_id")
+        return get_object_or_404(self.model, id=redirect_id)
 
-    if request.method == "POST":
-        form = RedirectForm(request.POST, request.FILES, instance=theredirect)
-        if form.is_valid():
-            with transaction.atomic():
-                form.save()
-                log(instance=theredirect, action="wagtail.edit")
-            messages.success(
-                request,
-                _("Redirect '%(redirect_title)s' updated.")
-                % {"redirect_title": theredirect.title},
-                buttons=[
-                    messages.button(
-                        reverse("wagtailredirects:edit", args=(theredirect.id,)),
-                        _("Edit"),
-                    )
-                ],
-            )
-            return redirect("wagtailredirects:index")
-        else:
-            messages.error(request, _("The redirect could not be saved due to errors."))
```

Comment:
Same here, we could set `error_message` on the view class instead:

```suggestion
    error_message = gettext_lazy("The redirect could not be saved due to errors.")
```